x/evm/wrappers/bank_test.go

```diff
@@ -0,0 +1,752 @@
+						gomock.Any(),
+						evmtypes.ModuleName,
+						sdk.AccAddress([]byte("test_address")),
+						expectedCoins,
+					).Return(nil)
+			},
+		},
+		{
+			name:        "success - 18 decimals amount not modified",
+			evmDenom:    TokenDenom,
+			evmDecimals: 18,
+			amount:      big.NewInt(1e18), // 1 token in 18 decimals
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin(TokenDenom, sdkmath.NewInt(1e18))
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(nil)
+
+				mbk.EXPECT().
+					SendCoinsFromModuleToAccount(
+						gomock.Any(),
+						evmtypes.ModuleName,
+						sdk.AccAddress([]byte("test_address")),
+						expectedCoins,
+					).Return(nil)
+			},
+		},
+		{
+			name:        "fail - mint coins error",
+			evmDenom:    TokenDenom,
+			evmDecimals: 6,
+			amount:      big.NewInt(1e18),
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "failed to mint coins to account in bank wrapper",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin(TokenDenom, sdkmath.NewInt(1e6))
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(errors.New("mint error"))
+			},
+		},
+		{
+			name:        "fail - send coins error",
+			evmDenom:    "evm",
+			evmDecimals: 6,
+			amount:      big.NewInt(1e18),
+			recipient:   sdk.AccAddress([]byte("test_address")),
+			expectErr:   "send error",
+			mockSetup: func(mbk *testutil.MockBankWrapper) {
+				expectedCoin := sdk.NewCoin("evm", sdkmath.NewInt(1e6))
+				expectedCoins := sdk.NewCoins(expectedCoin)
+
+				mbk.EXPECT().
+					MintCoins(gomock.Any(), evmtypes.ModuleName, expectedCoins).
+					Return(nil)
+
+				mbk.EXPECT().
+					SendCoinsFromModuleToAccount(
+						gomock.Any(),
+						evmtypes.ModuleName,
+						sdk.AccAddress([]byte("test_address")),
+						expectedCoins,
+					).Return(errors.New("send error"))
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Setup EVM configurator to have access to the EVM coin info.
+			configurator := evmtypes.NewEVMConfigurator()
+			configurator.ResetTestConfig()
+			err := configurator.WithEVMCoinInfo(tc.evmDenom, tc.evmDecimals).Configure()
+			require.NoError(t, err, "failed to configure EVMConfigurator")
+
+			// Setup mock controller
+			ctrl := gomock.NewController(t)
+
+			mockBankKeeper := testutil.NewMockBankWrapper(ctrl)
+			tc.mockSetup(mockBankKeeper)
+
+			bankWrapper := wrappers.NewBankWrapper(mockBankKeeper)
+			err = bankWrapper.MintAmountToAccount(context.Background(), tc.recipient, tc.amount)
+
+			if tc.expectErr != "" {
+				require.ErrorContains(t, err, tc.expectErr)
+			} else {
+				require.NoError(t, err)
+			}
```

Comment:
why are we using mocking here instead of the usual testing utils? Testing with the full suite would be better or not?